apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -105,8 +110,38 @@ class VisualizerModal extends React.Component {
     return columns.filter(column => isColumnNumeric(records, column));
   });
 
+  getValuesForFilterColumn = memoize((records, column) => {
+    if (!records || !column) {
+      return [];
+    }
+    let values = Array.from(new Set(records.map(record => record[column])));
+    return sortAlphabeticallyOrNumerically(values);
+  });
+
+  filterRecords = memoize((records, column, value) => {
+    if (!records) {
+      return [];
+    }
+    if (!column || !value) {
+      return records;
+    }
+    if (isNumber(value)) {
+      value = parseFloat(value);
+    }
+    return records.filter(record => record[column] === value);
+  });
+
   render() {
     const parsedRecords = this.parseRecords(this.props.tableRecords);
+    let filteredRecords = parsedRecords;
+    if (this.state.filterColumn && this.state.filterValue) {
```

Comment:
You're checking whether these are provided in both your memoized function, above, and before calling it, here.  I'd do one or the other.  Also, same comment about falsy values.